python/tskit/tables.py

```diff
@@ -3451,6 +3451,44 @@ def union(
             )
 
     def find_ibd(self, samples, max_time=None, min_length=None):
+        """
+        Returns a dictionary containing all IBD (identity-by-descent) segments for the
+        supplied list of sample pairs.
+
+        A pair of nodes ``(u, v)`` has an IBD segment with a left and right coordinate
+        ``[left, right)`` and ancestral node ``a`` iff the most recent common ancestor
+        of the segment ``[left, right)`` in nodes ``u`` and ``v`` is ``a``, and the
+        segment has been inherited along the same genealogical path (ie. it has not
+        been broken by recombination). The segments returned are the longest possible
+        ones.
+
+        Note that this definition is purely genealogical -- allelic states
+        *are not* considered here. If used without time or length thresholds, the
+        segments returned for a given pair will partition the span of the contig
+        represented by the tree sequence.
+
+        :param list samples: A list of pairs of integers, with each pair specified as
+            a tuple. In each pair, the integers correspond to the node IDs of a pair of
+            chromosomes whose IBD segments will be returned by
+            :meth:`.TableCollection.find_ibd`.
+        :param float max_time: Only segments inherited from common
+            ancestors whose node times are more recent than the specified time
+            will be returned. Specifying a maximum time is strongly recommended when
+            working with large tree sequences.
+        :param float min_length: Only segments longer than the specified length
+            will be returned.
+        :return: A dictionary object indexed by the specified sample pairs. The value
+            of each item is itself a dictionary indexed by the keys ``left``, ``right``
+            and ``node``. If the outputted object is saved to a variable called ``out``
+            and ``(u, v)`` is a specified sample pair, then
```

Comment:
Rather than putting this detail here in the return block, I wonder if it would be better to have another paragraph above in the description where we describe the output (maybe giving an example of the output?), and reserve this for describing the types. We should probably be precise about the dtypes as well (float64, and int32).